bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/DahuaHandler.java

```diff
@@ -352,6 +437,16 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             .sendHttpGET("/cgi-bin/configManager.cgi?action=setConfig&MotionDetect[0].Enable=false");
                 }
                 return;
+            case CHANNEL_MOTION_DETECTION_LEVEL:
+                int motionlevel = Math.round(Float.valueOf(command.toString()));
+
+                if (motionlevel == 0) {
+                    ipCameraHandler.sendHttpGET("/cgi-bin/configManager.cgi?action=setConfig&MotionDetect[0].Level=1");
+                } else {
+                    ipCameraHandler.sendHttpGET(
+                            "/cgi-bin/configManager.cgi?action=setConfig&MotionDetect[0].Level=" + motionlevel);
+                }
```

Comment:
```suggestion
if (command instanceof PercentType) {
if (PercentType.ZERO.equals(command)) {
ipCameraHandler.sendHttpGET("/cgi-bin/configManager.cgi?action=setConfig&MotionDetect[0].Level=1");
}else{
ipCameraHandler.sendHttpGET(
                            "/cgi-bin/configManager.cgi?action=setConfig&MotionDetect[0].Level=" + ((PercentType) command).intValue());
}
}
```

Please test and use this code instead of yours. This is much more CPU efficient and also will not create issues if someone does not send a slider value to the channel. PercentType is a BigDecimal internally, so it is very simple to get it as an Integer. If someone was to send a String command, my guess is your code would fail with an error of some kind.